SharedSupport/Default Bundles/CopyasMarkdown.saBundle/command.plist

```diff
@@ -3,45 +3,34 @@
 <plist version="1.0">
 <dict>
 	<key>author</key>
-	<string>David Buxton</string>
+	<string>Christian Nadolle</string>
+	<key>bundleVersion</key>
+	<integer>2</integer>
 	<key>category</key>
 	<string>Copy</string>
 	<key>command</key>
-	<string>#!/usr/bin/python2.7
-
-import os
-import subprocess
-import sys
-
-# Force pasteboard to treat this text as UTF-8 encoded.
-os.environ['LANG'] = 'en_US.UTF-8'
-
-rows = [line.strip().split('\t') for line in sys.stdin]
-longest = [max(len(v) for v in col) for col in zip(*rows)]
-out_rows = [' | '.join(v.ljust(l) for l, v in zip(longest, row)) + ' |' for row in rows]
-out_rows.insert(1, '|'.join('-' * (l + 2) for l in longest)[1:] + '|')
-out_value = '\n'.join(out_rows)
-
-proc = subprocess.Popen(['/usr/bin/pbcopy'], stdin=subprocess.PIPE)
-proc.communicate(input=out_value)
-</string>
+	<string>cat | /usr/bin/perl markdown-table.pl  | __CF_USER_TEXT_ENCODING=$UID:0x8000100:0x8000100 pbcopy</string>
 	<key>contact</key>
-	<string>qnivq@tnfznex6.pbz</string>
+	<string>tvg@udk.qr</string>
 	<key>description</key>
-	<string>Generates a text table in Markdown format. The text is sent to the pasteboard.</string>
+	<string>Copy Selected Result rows as Markdown Table. Uses extra time and memory to padd rows to maximum width. Blobs excluded.
+
+Version 1.0</string>
 	<key>input</key>
 	<string>selectedtablerowsastab</string>
+	<key>keyEquivalent</key>
+	<string></string>
 	<key>name</key>
-	<string>Copy as Markdown</string>
+	<string>Copy as Markdown (Padded)</string>
 	<key>scope</key>
 	<string>datatable</string>
 	<key>tooltip</key>
-	<string>Copy as Markdown</string>
+	<string>Copy Select as pretty Markdown Table </string>
```

Comment:
Maybe drop the "pretty" from this haha

Reply:
My first version used less memory and cycles by just printing a valid markdown table, but without padding columns with spaces to the same width (wich makes the markdown 'source code' look prettier). Therefore the "pretty" here ;-)
I did not test how much more memory is needed on huge datasets; maybe the 'lite' version would be interesting too, as alternative. What di you think?

Reply:
I reconsidered. It's a feature of markdown to be readable by humans -- in 'source' without passing it through a renderer -- so the padded table should be the default and only one.
Removed the "pretty" and "padded" stuff from the description and name.